test/metabase/driver/mysql_test.clj

```diff
@@ -856,46 +856,85 @@
-
-                    (sync/sync-database! database)
-                    (is (= {"readonly_table"   false
-                            "readwrite_table"  false
-                            "fullaccess_table" true}
-                           (t2/select-fn->fn :name :is_writable :model/Table :db_id (:id database)))))))
-
-              (finally
-                (jdbc/execute! spec "DROP USER IF EXISTS 'sync_writable_test_user';"))))))))
+                  (is (= {"readonly_table"   true
+                          "readwrite_table"  true
+                          "fullaccess_table" true}
+                         (t2/select-fn->fn :name :is_writable :model/Table :db_id (:id database)))))))
+            (finally
+              (jdbc/execute! spec "DROP USER IF EXISTS 'sync_writable_test_user';"))))))))
+
+(deftest partial-revokes-writable-test
+  (mt/test-driver :mysql
+    (when-not (mysql/mariadb? (mt/db))
+      (testing "`database-supports :check-table-writable` returns true normally but false with partial revokes"
+        (tx/drop-if-exists-and-create-db! driver/*driver* "partial_revokes_test")
+        (let [details (tx/dbdef->connection-details :mysql :db {:database-name "partial_revokes_test"})
+              spec    (sql-jdbc.conn/connection-details->spec :mysql details)]
+          (try
+            ;; Create test tables and user
+            (doseq [stmt ["CREATE TABLE `test_table` (id INTEGER);"
+                          "CREATE USER 'partial_revokes_test_user' IDENTIFIED BY 'password';"
+                          "GRANT SELECT, INSERT, UPDATE, DELETE ON partial_revokes_test.test_table TO 'partial_revokes_test_user'"]]
+              (jdbc/execute! spec stmt))
+
+            (let [user-connection-details (assoc details
+                                                 :user "partial_revokes_test_user"
+                                                 :password "password"
+                                                 :ssl true
+                                                 :additional-options "trustServerCertificate=true")]
+              (mt/with-temp [:model/Database database {:engine "mysql", :details user-connection-details}]
+                (testing "With partial_revokes OFF (default), check-table-writable is supported"
+                  (jdbc/execute! spec "SET GLOBAL partial_revokes = OFF;")
+                  (is (true? (driver/database-supports? driver/*driver* :check-table-writable database))
+                      "Should support check-table-writable when partial_revokes is OFF"))
+
+                (testing "With partial_revokes ON, check-table-writable is not supported"
+                  (jdbc/execute! spec "SET GLOBAL partial_revokes = ON;")
+                  (is (false? (driver/database-supports? driver/*driver* :check-table-writable database))
+                      "Should not support check-table-writable when partial_revokes is ON"))
```

Comment:
How about also testing that `is_writable` syncs to `nil` for this and the next case?